### PR TITLE
Add user-scoping permissions

### PR DESCRIPTION
**Pull request by microzon**
User-scoping permissions are for getting DataLossPrevention policies and InformationProtection configurations for a specific user.

### CODE REVIEW

**Comment by microzon** [comment]
> @microsoft-github-policy-service agree company="Microsoft"

@microsoft-github-policy-service agree company="Microsoft"

**Comment by FaithOmbongi** [comment]
Hi @microzon - Please follow the [Get started steps](https://dev.azure.com/msazure/One/_wiki/wikis/Microsoft%20Graph%20Partners/211776/Get-started?anchor=get-started-steps) to set yourself up as an internal contributor. Takes ~2mins.

@Danielabom - Can you pick this up for review? Thanks!

File: concepts/permissions-reference.md [hunk 1]
```diff
@@ -703,6 +703,22 @@ For more complex scenarios involving multiple permissions, see [Permission scena
 
 ---
 
+## Data Loss Prevention permissions
+
+#### Delegated permissions
+
+|   Permission    |  Display String   |  Description | Admin Consent Required |
+|:----------------|:------------------|:-------------|:-----------------------|
+| _DataLossPrevention.Read_ | Read data loss prevention policies applicable to a user | Allows an app to read data loss prevention policies, on behalf of the signed-in user. | Yes |
```
Comment (Danielabom) [inline concepts/permissions-reference.md:712]:
```suggestion
| _DataLossPrevention.Read_ | Read data loss prevention policies applicable to a user | Allows an app to read data loss prevention policies on behalf of the signed-in user. | Yes |
```

File: concepts/permissions-reference.md [hunk 2]
```diff
@@ -1227,6 +1243,22 @@ Incidents permissions are valid only on work or school accounts.
 
 ---
 
+## Information Protection Configuration permissions
+
+#### Delegated permissions
+
+|   Permission    |  Display String   |  Description | Admin Consent Required |
+|:----------------|:------------------|:-------------|:-----------------------|
+| _InformationProtectionConfig.Read_ | Read information protection configurations applicable to a user | Allows an app to read information protection configurations, on behalf of the signed-in user. | Yes |
```
Comment (Danielabom) [inline concepts/permissions-reference.md:1252]:
```suggestion
| _InformationProtectionConfig.Read_ | Read information protection configurations applicable to a user | Allows an app to read information protection configurations on behalf of the signed-in user. | Yes |
```

**Review by Danielabom — CHANGES_REQUESTED** [review]
Thank you for working on this PR @microzon. Please review my comments: 
We need a changelog entry for each new permission. See example 14 from this link:  [(https://dev.azure.com/msazure/One/_wiki/wikis/Microsoft%20Graph%20Partners/211742/Changelog)
I need to confirm somewhere the APi topics that will use these permissions, like an Api.md file or a roadmap item.

**Comment by microzon** [comment]
> Thank you for working on this PR @microzon. Please review my comments: We need a changelog entry for each new permission. See example 14 from this link: [(https://dev.azure.com/msazure/One/_wiki/wikis/Microsoft%20Graph%20Partners/211742/Changelog) I need to confirm somewhere the APi topics that will use these permissions, like an Api.md file or a roadmap item.

@Danielabom : The api.md where the APIs that will use the new permissions are defined: https://microsoftgraph.visualstudio.com/onboarding/_git/onboarding?path=%2Freviews%2F213-Add-data-classification-service-API-to-Graph%2Fapi.md&version=GBmaster&_a=preview.

Specific PRs for adding the 2 new APIs:
1. https://microsoftgraph.visualstudio.com/onboarding/_git/onboarding/pullrequest/11857
2. https://microsoftgraph.visualstudio.com/onboarding/_git/onboarding/pullrequest/11990

File: changelog/Microsoft.DataClassificationService.json [hunk 3]
```diff
@@ -0,0 +1,46 @@
+{
+    "changelog": [
+    {
+      "ChangeList": [
+        {
+          "Id": "14f49b9f-4bf2-4d24-b80e-b27ec58409bd",
+          "ApiChange": "Permission",
+          "ChangedApiName": "DataLossPrevention.Read.All",
+          "ChangeType": "Addition",
+          "Description": "Added support for the application permission of `DataLossPrevention.Read.All` to get the data loss prevention policies for the user resource.",
```
Comment (Danielabom) [inline changelog/Microsoft.DataClassificationService.json:10]: Hi, please follow the description format listed in example 14 of this wiki [https://dev.azure.com/msazure/One/_wiki/wikis/Microsoft%20Graph%20Partners/211742/Changelog]. 

 "Description": "Added support for the policy-based application permission of `DataLossPrevention.Read.All` to the read operations of the [_**_resource_**_](https://learn.microsoft.com/en-us/graph/api/resources/**_resourcename_**?view=graph-rest-**_version_**) resource.",

File: changelog/Microsoft.DataClassificationService.json [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+          "Target": "user"
+        },
+        {
+          "Id": "14f49b9f-4bf2-4d24-b80e-b27ec58409bd",
+          "ApiChange": "Permission",
+          "ChangedApiName": "DataLossPrevention.Read",
+          "ChangeType": "Addition",
+          "Description": "Added support for the delegated permission of `DataLossPrevention.Read` to get the data loss prevention policies for the user resource.",
```
Comment (Danielabom) [inline changelog/Microsoft.DataClassificationService.json:18]: Hi, please follow the description format listed in example 14 of this wiki [https://dev.azure.com/msazure/One/wiki/wikis/Microsoft%20Graph%20Partners/211742/Changelog].
You are missing the resource links; check this example:
"Added support for the policy-based application permission of OnlineMeetings.ReadWrite.All to the read and write operations of the resource name (_resource link__) resource.",

File: changelog/Microsoft.DataClassificationService.json [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+          "ChangeType": "Addition",
+          "Description": "Added support for the application permission of `InformationProtectionConfig.Read.All` to get the information protection configurations for the user resource.",
+          "Target": "user"
+        },
+        {
+          "Id": "14f49b9f-4bf2-4d24-b80e-b27ec58409bd",
+          "ApiChange": "Permission",
+          "ChangedApiName": "InformationProtectionConfig.Read",
+          "ChangeType": "Addition",
+          "Description": "Added support for the delegated permission of `InformationProtectionConfig.Read` to get the information protection configurations for the user resource.",
```
Comment (Danielabom) [inline changelog/Microsoft.DataClassificationService.json:34]: Hi, please follow the description format listed in example 14 of this wiki [https://dev.azure.com/msazure/One/wiki/wikis/Microsoft%20Graph%20Partners/211742/Changelog].
You are missing the resource links; check this example:
"Added support for the policy-based application permission of OnlineMeetings.ReadWrite.All to the read and write operations of the resource name (_resource link__) resource.",

**Review by Danielabom — CHANGES_REQUESTED** [review]
Please review my comments. Thanks. 
By the way, the GS tool is not expected to automatically create changelog entries for new permissions. Those are always added manually.

File: changelog/Microsoft.DataClassificationService.json (hunk 3 above)
Comment (microzon) [reply changelog/Microsoft.DataClassificationService.json]: Updated as suggested.

File: changelog/Microsoft.DataClassificationService.json (hunk 4 above)
Comment (microzon) [reply changelog/Microsoft.DataClassificationService.json]: Updated as suggested.

File: changelog/Microsoft.DataClassificationService.json (hunk 5 above)
Comment (microzon) [reply changelog/Microsoft.DataClassificationService.json]: Updated as suggested.

File: concepts/permissions-reference.md [hunk 6]
```diff
@@ -1227,6 +1243,22 @@ Incidents permissions are valid only on work or school accounts.
 
 ---
 
+## Information Protection Configuration permissions
+
+#### Delegated permissions
+
+|   Permission    |  Display String   |  Description | Admin Consent Required |
+|:----------------|:------------------|:-------------|:-----------------------|
+| _InformationProtectionConfig.Read_ | Read information protection configurations applicable to a user | Allows an app to read information protection configurations on behalf of the signed-in user. | Yes |
```
Comment (Danielabom) [inline concepts/permissions-reference.md:1252]:
```suggestion
| _InformationProtectionConfig.Read_ | Read information protection configurations applicable to a user | Allows an app to read information protection configurations on behalf of the signed-in user. | Yes|
```

**Review by Danielabom — APPROVED** [review]
Changes look good. Thanks, @microzon

**Comment by Danielabom** [comment]
Contacted the author via email to follow-up.

**Comment by microzon** [comment]
Closing the pull request as Graph team suggested to rename "DataLossPrevention.Read" due to hash collision with MSA permission.

**Pull request closed without merging**